ios/AppcuesReactNative.swift

```diff
@@ -73,8 +73,10 @@ class AppcuesReactNative: RCTEventEmitter {
     }
 
     @objc
-    func show(_ experienceID: String) {
-        implementation?.show(experienceID: experienceID)
+    func show(_ experienceID: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
+        implementation?.show(experienceID: experienceID) { success, _ in
+            resolve(success)
```

Comment:
Should we be calling `reject` when `success == false`?

Reply:
im not really sure, but based this on `didHandleURL` below, where it only uses `resolve`, including the error cases.

Reply:
looks like `reject` would actually throw an error https://gist.github.com/chourobin/f83f3b3a6fd2053fad29fff69524f91c#file-promises-md

Reply:
I'd think an error the right thing in this context? Probably worth checking with a few of our many JS experts on what they'd prefer to work with.

I'd argue the `didHandleURL` is different because a boolean `false` is the correct state of "did it handle the URL?", versus this saying "show something" and then throwing an error if it fails.

Reply:
I think I'd agree with you in theory, but might be less straightforward in practice, the way it is implemented across platforms.  A way it can fail hypothetically is an invalid experience ID - network lookup fails.  On Android, this is a suspend function that returns a boolean - success or failure. On iOS, it is a callback with a boolean AND an error.  iOS would provide a `false` value AND the error in that case, while android would provide a `false` value.  I wouldn't necessarily say Android is better by any means, but we don't currently have the information to provide an accurate error message beyond the same "it did not succeed" boolean value we have here.

Reply:
What if we used a single hardcoded error message? That would at least maintain the expected promise interface, even if the error message isn't helpful.

Reply:
updated - now rejects on error, also returns a promise but is not async in the method signature.